frontend/src/oso-extensions/oso-notebook-app.tsx

```diff
@@ -0,0 +1,189 @@
+/* Copyright 2024 Marimo. All rights reserved. */
+
+import { usePrevious } from "@dnd-kit/utilities";
+import { TooltipProvider } from "@radix-ui/react-tooltip";
+import { useAtomValue, useSetAtom } from "jotai";
+import { useEffect } from "react";
+import { Controls } from "@/components/editor/controls/Controls";
+import { AppHeader } from "@/components/editor/header/app-header";
+import { FilenameForm } from "@/components/editor/header/filename-form";
+import { MultiCellActionToolbar } from "@/components/editor/navigation/multi-cell-action-toolbar";
+import { cn } from "@/utils/cn";
+import { Paths } from "@/utils/paths";
+import { AppContainer } from "../components/editor/app-container";
+import {
+  useRunAllCells,
+  useRunStaleCells,
+} from "../components/editor/cell/useRunCells";
+import { CellArray } from "../components/editor/renderers/CellArray";
+import { CellsRenderer } from "../components/editor/renderers/cells-renderer";
+import { useHotkey } from "../hooks/useHotkey";
+import {
+  cellIdsAtom,
+  hasCellsAtom,
+  notebookIsRunningAtom,
+  numColumnsAtom,
+  useCellActions,
+} from "../core/cells/cells";
+import { CellEffects } from "../core/cells/effects";
+import type { AppConfig, UserConfig } from "../core/config/config-schema";
+import { RuntimeState } from "../core/kernel/RuntimeState";
+import { getSessionId } from "../core/kernel/session";
+import { useTogglePresenting } from "../core/layout/useTogglePresenting";
+import { viewStateAtom } from "../core/mode";
+import { useRequestClient } from "../core/network/requests";
+import { useFilename } from "../core/saving/filename";
+import { lastSavedNotebookAtom } from "../core/saving/state";
+import { useJotaiEffect } from "../core/state/jotai";
+import { useMarimoWebSocket } from "../core/websocket/useMarimoWebSocket";
+
+interface AppProps {
+  /**
+   * The user config.
+   */
+  userConfig: UserConfig;
+  /**
+   * The app config.
+   */
+  appConfig: AppConfig;
+  /**
+   * If true, the floating controls will be hidden.
+   */
+  hideControls?: boolean;
+}
+
+export const OSONotebookApp: React.FC<AppProps> = ({
```

Comment:
This isn't currently in use, but leaving this here for now as we will likely want to override this completely.